blocks/api/raw-handling/utils.js

```diff
@@ -1,174 +1,91 @@
 /**
  * External dependencies
  */
-import { includes } from 'lodash';
+import { omit, mergeWith } from 'lodash';
 
 /**
- * Browser dependencies
+ * WordPress dependencies
  */
-const { ELEMENT_NODE, TEXT_NODE } = window.Node;
+import { unwrap, insertAfter, remove } from '@wordpress/utils';
 
 /**
- * An array of tag groups used by isInlineForTag function.
- * If tagName and nodeName are present in the same group, the node should be treated as inline.
- * @type {Array}
+ * Browser dependencies
  */
-const inlineWhitelistTagGroups = [
-	[ 'ul', 'li', 'ol' ],
-	[ 'h1', 'h2', 'h3', 'h4', 'h5', 'h6' ],
-];
-
-const inlineWhitelist = {
-	strong: {},
-	em: {},
-	del: {},
-	ins: {},
-	a: { attributes: [ 'href' ] },
-	code: {},
-	abbr: { attributes: [ 'title' ] },
-	sub: {},
-	sup: {},
-	br: {},
-};
-
-const embeddedWhiteList = {
-	img: { attributes: [ 'src', 'alt' ], classes: [ 'alignleft', 'aligncenter', 'alignright', 'alignnone' ] },
-	iframe: { attributes: [ 'src', 'allowfullscreen', 'height', 'width' ] },
-};
-
-const inlineWrapperWhiteList = {
-	figcaption: {},
-	h1: {},
-	h2: {},
-	h3: {},
-	h4: {},
-	h5: {},
-	h6: {},
-	p: {},
-	li: { children: [ 'ul', 'ol', 'li' ] },
-	pre: {},
-	td: {},
-	th: {},
-};
-
-const whitelist = {
-	...inlineWhitelist,
-	...inlineWrapperWhiteList,
-	...embeddedWhiteList,
-	figure: {},
-	blockquote: {},
-	hr: {},
-	ul: {},
-	ol: { attributes: [ 'type' ] },
-	table: {},
-	thead: {},
-	tfoot: {},
-	tbody: {},
-	tr: {},
-};
-
-export function isWhitelisted( element ) {
-	return whitelist.hasOwnProperty( element.nodeName.toLowerCase() );
-}
-
-export function isNotWhitelisted( element ) {
-	return ! isWhitelisted( element );
-}
-
-export function isAttributeWhitelisted( tag, attribute ) {
-	return (
-		whitelist[ tag ] &&
-		whitelist[ tag ].attributes &&
-		whitelist[ tag ].attributes.indexOf( attribute ) !== -1
-	);
-}
+const { ELEMENT_NODE, TEXT_NODE } = window.Node;
 
 /**
- * Checks if nodeName should be treated as inline when being added to tagName.
- * This happens if nodeName and tagName are in the same group defined in inlineWhitelistTagGroups.
+ * Get schema of possible paths for phrasing content.
  *
- * @param {string} nodeName Node name.
- * @param {string} tagName  Tag name.
+ * @see https://developer.mozilla.org/en-US/docs/Web/Guide/HTML/Content_categories#Phrasing_content
  *
- * @return {boolean} True if nodeName is inline in the context of tagName and
- *                    false otherwise.
+ * @return {Object} Schema.
  */
-function isInlineForTag( nodeName, tagName ) {
-	if ( ! tagName || ! nodeName ) {
-		return false;
-	}
-	return inlineWhitelistTagGroups.some( ( tagGroup ) =>
-		includes( tagGroup, nodeName ) && includes( tagGroup, tagName )
-	);
-}
-
-export function isInline( node, tagName ) {
-	const nodeName = node.nodeName.toLowerCase();
-	return inlineWhitelist.hasOwnProperty( nodeName ) || isInlineForTag( nodeName, tagName );
-}
+export function getPhrasingContentSchema() {
```

Comment:
How often do we call this? Does it need to be a function, or can we construct this (or at least parts of it, like the base schema) as a constant?

Reply:
Good point! I wrapped it in a function so that it cannot be changed by block authors for the whole application. I guess we can calculate once and return the constant.